File: wun-android/src/main/kotlin/wun/foundation/TextRenderer.kt
// Compose renderer for `:wun/Text`. `:variant` chooses h1 / h2 /
// body typography presets.

package wun.foundation

import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.sp
import kotlinx.serialization.json.JsonPrimitive
import wun.WunComponent

object TextRenderer {
    val render: WunComponent = { props, children ->
        val variant = (props["variant"] as? JsonPrimitive)?.let {
            if (it.isString) it.content else null
        }
        val text = Children.flatten(children)
        val style: TextStyle = when (variant) {
            "h1" -> TextStyle(fontSize = 32.sp, fontWeight = FontWeight.SemiBold)
            "h2" -> TextStyle(fontSize = 22.sp, fontWeight = FontWeight.SemiBold)
            else -> TextStyle(fontSize = 15.sp)
        }
        Text(text, style = style)
    }
}

File: wun-ios/Sources/Wun/Foundation/TextRenderer.swift
// SwiftUI renderer for `:wun/Text`. `:variant` chooses a typography
// preset (h1 / h2 / body); children that are strings are joined to
// form the rendered text.

import SwiftUI

public enum WunText {
    public static let render: WunComponent = { props, children in
        let variant = props["variant"]?.stringValue ?? "body"
        let text    = WunChildren.flatten(children)

        switch variant {
        case "h1":
            return AnyView(SwiftUI.Text(text)
                .font(.system(size: 32, weight: .semibold)))
        case "h2":
            return AnyView(SwiftUI.Text(text)
                .font(.system(size: 22, weight: .semibold)))
        default:
            return AnyView(SwiftUI.Text(text)
                .font(.system(size: 15))
                .opacity(0.85))
        }
    }
}
